src/factories/Axis.ts

```diff
@@ -38,7 +44,7 @@ module n3Charts.Factory {
       this.updateScaleDomain(extent);
 
       this.axis = this.getAxis(this.scale, axisOptions);
-      this.updateAxisOrientation();
+      this.updateAxisOrientation(this.axis);
```

Comment:
Great, in general I prefer this way over accessing the variables via the `this` keyword.